### ISSUE TITLE
Make Ameland and IABR scenarios available in the start menu

### ISSUE THREAD

**Issue by dennisquintel** [quintel/etmodel#2115]
See quintel/etengine#835

Comment by @grdw:

> The production scenarios are:
> IABR: https://pro.energytransitionmodel.com/scenarios/291793
> Ameland: https://pro.energytransitionmodel.com/scenarios/291787
> 
> Both scenarios belong to Alex's user.

including @AlexanderWirtz, @jorisberkhout

**Comment by dennisschoenmakers** [comment]
I think it makes sense to create a new header there called 'regional scenarios'

**Comment by grdw** [comment]
![screen shot 2016-05-20 at 09 44 53](https://cloud.githubusercontent.com/assets/2676542/15420907/87706302-1e6f-11e6-868b-63be7c8e09bb.png)

Just to be clear, you mean in here?

**Comment by dennisschoenmakers** [comment]
Yes, in the menu! I Looking at it now, I think you can put them in the
'Features' menu. Two scenario's: Ameland & the IABR 2035 please. :)

On Fri, May 20, 2016 at 9:45 AM, Gerard <notifications@github.com> wrote:

> [image: screen shot 2016-05-20 at 09 44 53]
> <https://cloud.githubusercontent.com/assets/2676542/15420907/87706302-1e6f-11e6-868b-63be7c8e09bb.png>
>
> Just to be clear, you mean in here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/quintel/etmodel/issues/2115#issuecomment-220539732>
>



-- 

Dennis Schoenmakers, *Partner*


*Quintel Intelligence*Atrium - Strawinskylaan 3051
1077 ZX  Amsterdam
The Netherlands

*Visiting Address:*
Keizersgracht 639
1017 DT Amsterdam

T  : +31 (0)20 301 22 69
F  : +31 (0)20 301 22 02
M : +31 (0)6 109 65 293

E : dennis.schoenmakers@quintel.com
I  : www.quintel.com and www.energytransitionmodel.com

*This message (including any attachments) is confidential and may be
privileged. Any review, retransmission, dissemination or other use of, or
taking any action with reference to this information by persons other than
the intended recipient is strictly prohibited. If you have received it by
mistake please notify the sender by return e-mail and delete this message
from your system. Please note that e-mails are susceptible to change. The
sender will not accept liability for the improper or incomplete
transmission of the information contained in this message.*

**Comment by grdw** [comment]
If I understand correctly the scenarios in that menu are presets from ETEngine right? ( @antw ). So in order for me to create two new one's I have to create two new presets in etsource in order to add them to the menu.

**Comment by jorisberkhout** [comment]
See also https://github.com/quintel/etsource/issues/1109

**Comment by grdw** [comment]
> See also quintel/etsource#1109

@dennisschoenmakers makers famous last words indeed ... 

I could create the presets but not scaled. Can't we just create a simple link to these scenarios in that part of the menu? Or would that be a little against the purpose of existing scenarios?

**Comment by dennisschoenmakers** [comment]
> I could create the presets but not scaled. Can't we just create a simple link to these scenarios in that part of the menu? Or would that be a little against the purpose of existing scenarios?

I think that is the most logical choice here. Let's deal with creating scaled presets later in quintel/etengine#842.

**Comment by dennisschoenmakers** [comment]
Could we give this some priority @grdw ? 

cc @DorinevanderVlies

**Comment by grdw** [comment]
Well yes I could, but I have a strong preference to first fix https://github.com/quintel/etengine/issues/842 before starting to create a really hack a link in a menu.

**Comment by grdw** [comment]
I manage to find a way:

````ruby
  def scenario_groups(groups)
    groups.map do |group|
      group[:scenarios] += (custom_links(group[:name]) || [])
      group
    end
  end

  def custom_links(name)
    [
      Api::Scenario.new(title: 'Custom link', name: 'Custom link', id: 515515, display_group: "Actueel"),
      Api::Scenario.new(title: 'Custom link 2', name: 'Custom link 2, id: 515515, display_group: "Actueel")
    ].group_by(&:display_group)[name]
  end
````

*I do know that you can also do: `Api::Scenario.find(<scenario_id>)`, however I can't set the `display_group` attribute, that's why I'm creating new scenario Objects and fake the scenario id. 

With these two helpers I am able to create custom links for scaled scenarios in the ETModel interface. Maybe we could wrap it in a `*.yml` configuration file? Also in order to create title for both `en` and `nl`? Something like:

`custom_sidebar_links.yml`
````Yaml
---
custom_sidebar_links:
  - title: Custom link
    id: 515515
    display_group: "Actueel"
    key: "custom_link"
  - title: Custom link #2
    id: 515515
    display_group: "Actueel #2"
    key: "custom_link_2"
````

And use a translations file for the custom_links + the `key` attributes to initiate some custom description?

We could also turn it in a database table but I don't know how many custom links there will be created.

**Comment by grdw** [comment]
@dennisschoenmakers @antw does this sound like an ok solution?

**Comment by antw** [comment]
> does this sound like an ok solution?

I think so.

I'll take a look at etengine#842 to see how easy it would be; it would be a pity to implement this temporary fix if that issue turns out to be simple. I'll get back to your ASAP.

**Comment by antw** [comment]
The code is pushed and I have the two presets made and ready-to-go. However, the scenarios have the following validations errors:

* **ameland_2020_obv_2013_v9**
  * contains inputs belonging to the `district_heating_buildings_present` share group, but the values sum to 0.0, not 100
  * contains inputs belonging to the `district_heating_buildings` share group, but the values sum to 0.0, not 100

* **iabr_2035_stad_en_regio_groningen**
  * contains input keys which don't exist:
    * `industry_chemicals_other_burner_coal`
    * `transport_ship_using_biodiesel_share`
    * `transport_ship_using_diesel_share`
    * `transport_ship_using_heavy_fuel_oil_share`

I'm guessing the IABR messages are to be expected (#2251), and those inputs can be removed?

Ameland... I'm not sure about. It looks like both district heating share groups are missing a "heat network connection" input. Is it safe to set this value to 100?

* district_heating_buildings:

```
buildings_chp_engine_biogas_share = 0.0
buildings_collective_chp_network_gas_share = 0.0
buildings_collective_chp_wood_pellets_share = 0.0
buildings_collective_geothermal_share = 0.0
buildings_heat_network_connection_steam_hot_water_share = MISSING
```

* district_heating_buildings_present

```
buildings_chp_engine_biogas_share_present = 0.0
buildings_collective_chp_network_gas_share_present = 0.0
buildings_collective_chp_wood_pellets_share_present = 0.0
buildings_collective_geothermal_share_present = 0.0
buildings_heat_network_connection_steam_hot_water_share_present = MISSING
```

--

Ping: @AlexanderWirtz @ChaelKruip.

**Comment by dennisschoenmakers** [comment]
On Ameland: ping @DorinevdVlies

**Comment by dennisschoenmakers** [comment]
I meant @DorinevanderVlies

**Comment by grdw** [comment]
@ChaelKruip re-assign at will. I think somebody needs to look at those presets but I think that might already be done though.

**Comment by ChaelKruip** [comment]
> @ChaelKruip re-assign at will. I think somebody needs to look at those presets but I think that might already be done though.

I think @AlexanderWirtz and myself should have a look before we publish these presets.

**Comment by AlexanderWirtz** [comment]
What is the status of this? Close?

**Comment by ChaelKruip** [comment]
@AlexanderWirtz this is the moment to put this live. It looks as if only the points mentioned in https://github.com/quintel/etmodel/issues/2115#issuecomment-251443332 are keeping this from being deployed. Right @antw?

**Comment by DorinevanderVlies** [comment]
> Ameland... I'm not sure about. It looks like both district heating share groups are missing a "heat network connection" input. Is it safe to set this value to 100?

For Ameland the heat network is geothermal. So the input will be:

```
buildings_chp_engine_biogas_share = 0.0
buildings_collective_chp_network_gas_share = 0.0
buildings_collective_chp_wood_pellets_share = 0.0
buildings_collective_geothermal_share = 100.0
buildings_heat_network_connection_steam_hot_water_share = 0.0
```
- district_heating_buildings_present

```
buildings_chp_engine_biogas_share_present = 0.0
buildings_collective_chp_network_gas_share_present = 0.0
buildings_collective_chp_wood_pellets_share_present = 0.0
buildings_collective_geothermal_share_present = 100.0
buildings_heat_network_connection_steam_hot_water_share_present = 0.0
```

**Comment by antw** [comment]
Thanks @DorinevanderVlies.

I suspect the Ameland preset I created may be out-of-date. It came from production and was labelled as "v9", but production also has one labelled "1.0". Beta has versions of Ameland which are newer than the one on production:

<img width="762" alt="screen shot 2017-06-15 at 15 31 50" src="https://user-images.githubusercontent.com/4383/27186292-c7cd5942-51df-11e7-84fb-27d2e0db4999.png">

**Comment by DorinevanderVlies** [comment]
I think there are some things we should discuss here.

This Ameland project ran before we were able to make local dataset. So, there is no local dataset for Ameland.
We made a "scenario" that serves as a local dataset with end-year 2020.
We used that as a starting point to make 4 scenario's in total.

@dennisschoenmakers, what to which scenario do we want to link this preset? The current link is to the scenario that serves as a local dataset.

Two options:
- Link to the "scenario" that serves as a dataset? (Not my preference, in that case we should make a local dataset for Ameland) In that case I would link to scenario ID 545739 on the beta server. If this becomes a preset I have to change the title and description)
- Link to one of the 4 scenario's we made for Ameland (I would link to scenario ID 554943 on the beta server, "Ameland jaargemiddeld hernieuwbaar")
